public/app/plugins/panel/bargauge/BarGaugePanel.test.tsx

```diff
@@ -1,106 +1,134 @@
-import { mount, ReactWrapper } from 'enzyme';
+import { render, screen } from '@testing-library/react';
+import { uniqueId } from 'lodash';
 import React from 'react';
 
-import {
-  dateMath,
-  dateTime,
-  FieldConfigSource,
-  LoadingState,
-  PanelData,
-  PanelProps,
-  TimeRange,
-  toDataFrame,
-  VizOrientation,
-} from '@grafana/data';
+import { dateMath, dateTime, EventBus, LoadingState, TimeRange, toDataFrame, VizOrientation } from '@grafana/data';
 import { selectors } from '@grafana/e2e-selectors';
 import { BarGaugeDisplayMode } from '@grafana/ui';
 
-import { BarGaugePanel } from './BarGaugePanel';
-import { PanelOptions } from './models.gen';
+import { BarGaugePanel, BarGaugePanelProps } from './BarGaugePanel';
 
 const valueSelector = selectors.components.Panels.Visualization.BarGauge.valueV2;
 
 describe('BarGaugePanel', () => {
-  describe('when empty result is rendered', () => {
-    const wrapper = createBarGaugePanelWithData({
-      series: [],
-      timeRange: createTimeRange(),
-      state: LoadingState.Done,
-    });
+  describe('when there is no data', () => {
+    it('show a "No Data" message', () => {
+      const panelData = buildPanelData();
+
+      render(<BarGaugePanel {...panelData} />);
 
-    it('should render with title "No data"', () => {
-      const displayValue = wrapper.find(`div[data-testid="${valueSelector}"]`).text();
-      expect(displayValue).toBe('No data');
+      expect(screen.getByText(/no data/i)).toHaveTextContent('No data');
```

Comment:
```suggestion
      expect(screen.getByText('No data')).toBeInTheDocument();
```
Quite strange to use the same query to match and element and test for it. In this case it's better to test for presence in the DOM.